### ISSUE TITLE
trait-impl-incorrect-safety false-positive with unsafe fields

### ISSUE THREAD

**Issue by TimNN** [rust-lang/rust-analyzer#19477]
This requires the unstable `#![feature(unsafe_fields)]` (https://github.com/rust-lang/rust/issues/132922). (So feel free to close if this should wait until the feature is closer to stabilization).

Implementing `Copy` for a `struct` with an `unsafe` field requires an `unsafe impl` as of https://github.com/rust-lang/rust/pull/134008.

However, rust-analyzer reports an "unsafe impl for safe trait (trait-impl-incorrect-safety)" error for such an `impl`.

**rust-analyzer version**: `rust-analyzer version: 0.4.2359-standalone (2e1ff255ae 2025-03-28)`

**rustc version**: `rustc 1.87.0-nightly (3f5502370 2025-03-27)`

**editor or extension**: VSCode

**code snippet to reproduce**: 
```rust
#![feature(unsafe_fields)]
#![allow(incomplete_features)]

#[repr(transparent)]
pub struct Alpha { unsafe a: u8 }

impl Clone for Alpha {
    fn clone(&self) -> Self { *self }
}

unsafe impl Copy for Alpha {}
```

**Comment by K0bin** [comment]
I don't know if it's related but I'm getting a lot of false positive `trait-impl-incorrect-safety` errors for `unsafe impl Send for MyStruct {}` (and the equivalent Sync impl).

**Comment by ChayimFriedman2** [comment]
@K0bin shouldn't be related. Please open a separate issue.
